homeassistant/components/swidget/light.py

```diff
@@ -0,0 +1,77 @@
+"""Support for Swidget lights."""
+
+from __future__ import annotations
+
+import logging
+import math
+from typing import Any, cast
+
+from swidget.swidgetdimmer import SwidgetDimmer
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.color import brightness_to_value, value_to_brightness
+
+from . import SwidgetConfigEntry
+from .entity import CoordinatedSwidgetEntity
+
+_LOGGER = logging.getLogger(__name__)
+BRIGHTNESS = "brightness"
+BRIGHTNESS_SCALE = (0, 100)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: SwidgetConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up light."""
+    coordinator = config_entry.runtime_data
+    device = coordinator.device
+    if device.is_dimmer:
+        async_add_entities(
+            [SwidgetSmartDimmer(cast(SwidgetDimmer, device), coordinator)]
+        )
+
+
+class SwidgetSmartDimmer(CoordinatedSwidgetEntity, LightEntity):
+    """Representation of a Swidget Dimmer."""
+
+    device: SwidgetDimmer
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+    _attr_color_mode = ColorMode.BRIGHTNESS
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the light on."""
+        if (brightness := kwargs.get(ATTR_BRIGHTNESS)) is not None:
+            brightness = math.ceil(
+                brightness_to_value(BRIGHTNESS_SCALE, kwargs[ATTR_BRIGHTNESS])
+            )
```

Comment:
Is `ceil` really correct here, wouldn't `round` be better?

Reply:
I've updated. Please check.